### ISSUE TITLE
Rule Request: Duplicate Imports

### ISSUE THREAD

**Issue by jpsim** [realm/SwiftLint#1881]
### New Issue Checklist

- [x] Updated SwiftLint to the latest version
- [x] I searched for [existing GitHub issues](https://github.com/realm/SwiftLint/issues)

### Rule Request

> 1. Why should this rule be added? Share links to existing discussion about what
   the community thinks about this.

Duplicate imports are unnecessary.

> 2. Provide several examples of what _would_ and _wouldn't_ trigger violations.

```swift
// Should Trigger
import Foundation
import Dispatch
import Foundation

// Should Trigger
import Foundation
import Foundation.NSString

// Should Not Trigger
Import Foundation.NSString
Import Foundation.NSRange
```

> 3. Should the rule be configurable, if so what parameters should be configurable?

Shouldn't be configurable. Either there are duplicates or not.

> 4. Should the rule be opt-in or enabled by default? Why?
   See [README.md](../README.md#opt-in-rules) for guidelines on when to mark a
   rule as opt-in.

Enabled by default.

@marcelofabri removing this duplicate import motivated me to write this rule request: https://github.com/realm/SwiftLint/pull/1872/files#diff-71e28580a5fd3b8b3f701a7c8ca9f6a9

There are gotchas around conditional imports, so I suggest we just not count imports in a conditional compilation expression.

**Comment by marcelofabri** [comment]
Nice, I thought about filling an issue requesting this rule but I completely forgot later.

**Comment by marcelofabri** [comment]
Implemented in #2004
